GitUI/UserControls/RevisionGrid/Columns/CommitIdColumnProvider.cs

```diff
@@ -66,8 +67,26 @@ public override bool TryGetToolTip(DataGridViewCellMouseEventArgs e, GitRevision
                 return false;
             }
 
-            toolTip = revision.Guid;
+            toolTip = Format(revision.Guid);
             return true;
         }
+
+        public static string Format(string guid)
```

Comment:
After staring at this method for a bit I understand it formats the guid into four digit chunks separated by a space.
What escapes me is why it is needed. We don't format guids anywhere else in the app. Nor I haven't seen it is done anywhere else in relation to git repos.

No offence, but to me this method adds unnecessary complexity. It also negatively affects the performance as it will be run everytime and result in memory allocations.

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.guid.tostring?view=netframework-4.7.2#System_Guid_ToString_System_String_

Reply:
Yes, that's too. though because it is a string not a guid, we'll have to cast, that will lead to additional allocations.
In this case I don't believe it is necessary at all.

Reply:
This method is used only when a tooltip is to be displayed. This should have no substancial effect on the performance.

I cannot read such blobs of hexadecimal digits easily. It is a constant annoyance to me if they are longer than 6-7 digits.
To give some optical guiding, I inserted little gaps (about half a space).
You can see it on the screenshots. Give it a try, please!

A Git GUID is not a System.GUID (40 vs. 32 bits), @vbjay.

Reply:
My point was to make the git guid format correctly.  Define the formatting in the class.  You can even use format strings like System.GUID does. Then it will be as simple as something like this 

```c#
theGUID.ToString("D"); 
```

Reply:
> I cannot read such blobs of hexadecimal digits easily. It is a constant annoyance to me if they are longer than 6-7 digits.

Whilst I can empathise your feeling I can not justify the inconsistency in UX this change introduces. Hashes are universally rendered as one unbroken string - in GE, on GH, and other apps I've seen.
We had a discussion (#4420) whether we should dynamically calculate hash sizes, but in the end it was concluded to be a perf overhead.